SIL.WritingSystems/IcuUCharCategoryExtensions.cs

```diff
@@ -0,0 +1,77 @@
+using System.Globalization;
+using Icu;
+
+namespace SIL.WritingSystems
+{
+	public static class IcuUCharCategoryExtensions
+	{
+		public static UnicodeCategory ToUnicodeCategory(this Character.UCharCategory category)
+		{
+			switch (category)
+			{
+				case Character.UCharCategory.UNASSIGNED:
```

Comment:
Too bad that `Character.UCharCategory` and `UnicodeCategory` don't have the same values...